src/MessageTrait.php

```diff
@@ -170,8 +178,11 @@ public function withHeader($header, $value)
             );
         }
 
+        $lowerHeader = strtolower($header);
+
         $new = clone $this;
-        $new->headers[$header] = $value;
+        $new->headerNames[$lowerHeader] = $header;
+        $new->headers[$lowerHeader] = $value;
```

Comment:
@weierophinney Given you register the `X-Foo` & `X-foo` headers what should retrun `getHeader('x-foo')` and `getHeaders()`?